controllers/reconcile_test.go

```diff
@@ -495,7 +499,9 @@ func TestOpenTelemetryCollectorReconciler_Reconcile(t *testing.T) {
 					result: controllerruntime.Result{},
 					checks: []check[v1alpha1.OpenTelemetryCollector]{
 						func(t *testing.T, params v1alpha1.OpenTelemetryCollector) {
-							exists, err := populateObjectIfExists(t, &v1.ConfigMap{}, namespacedObjectName(naming.Collector(params.Name), params.Namespace))
+							configHash, _ := getConfigMapSHAFromString(params.Spec.Config)
```

Comment:
can you add a test into this reconcile file to ensure we clean up the old configmaps effectively? If you set the configmap versions to keep as one, then you can check that updating ensure that guarantee.

Reply:
if it's too much work let's just book a follow-up item given you have the e2e test. We're trying to rely more on these tests where we can as e2e testing can be a bit cumbersome.

Reply:
I started working on this but it seems like I either need to create a new test for v1beta1 objects (which is easy enough but duplicates a lot of boilerplate), or convert the existing reconcile test to v1beta1 which would be too much work. WDYT?

Reply:
ah let's just book a followup for this then, I will probs need to do that conversion :)

Reply:
https://github.com/open-telemetry/opentelemetry-operator/issues/2948